src/hardware/rdtech-dps/protocol.c

```diff
@@ -200,7 +202,7 @@ SR_PRIV int rdtech_dps_get_model_version(struct sr_modbus_dev_inst *modbus,
 		if (ret != SR_OK)
 			return ret;
 		rdptr = (void *)registers;
-		*model = read_u16be_inc(&rdptr) / 10;
+		*model = read_u16be_inc(&rdptr);
```

Comment:
can you elaborate on why this was dividing the model number by ten a model number, and why it's not necessary anymore ? Does that risk breaking anything ?

Reply:
Thank you for your question.

The old code used this kludge (dividing by 10) to extract the textual model designation from the actual numerical model number. This worked as long as there were numerical model numbers such as 60061, 60121, 60241. However the models 6006P and 6012P have the numerical model numbers 60065 and 60125. Dividing these by 10 throws away information such that one can no longer distinguish between e.g. 6006 and 6006P. In my mind, it is better to store the model number chosen by the manufacturer than to come up with an alternative scheme, especially if information is lost. This stored model number is anyway used to lookup the textual model designation from a table before displaying it in the GUI. This is the only use of the numerical model number in the old code that I'm aware of. In my PR, I, in addition, use the numerical model number to conditionally run some code.

So, no, I don't think this breaks anything. In fact, it makes the code more future safe, if Ruideng comes up with another new series of devices.

Reply:
Ok, fair. Was not aware of what the "model string" actually looked like.